services/121-service/src/payments/fsp-integration/commercial-bank-ethiopia/commercial-bank-ethiopia.mock.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class CommercialBankEthiopiaMockService {
+  public async waitForRandomDelay(): Promise<void> {
+    const min = 100;
+    const max = 300;
+    const randomNumber = Math.floor(Math.random() * (max - min + 1)) + min;
+    return new Promise((resolve) => setTimeout(resolve, randomNumber));
+  }
+
+  public async post(payload: any, payment): Promise<any> {
+    await this.waitForRandomDelay();
+    const response = {
+      Status: {
+        transactionId: { _text: 'FT212435G2ZD' },
+        messageId: {},
+        successIndicator: { _text: 'Success' },
+        application: { _text: 'FUNDS.TRANSFER' },
+        // messages: [
```

Comment:
@Vladimir-Petkov I see that you can uncomment this to mock the 'duplicate transaction' scenario, but it will then subsequently fail because the 'getTransactionStatus' does not have a mock version right? 

It would be really good to have a richer mock environment, so:
- also mock getTransactionStatus
- don't change the outcome of a mock-method by commenting/uncommenting some code, but by returning different responses based on some input parameter. See e.g. 'intersolve-visa-api-mock.service' where we base different outcomes on different lastNames. This then comes together with a specific test registration csv which is defined in 'test-registrations-OCW.csv' in the repo. Of course you would have to see what input parameter makes sense in the CBE-context (it should already be part of the payload you're passing)
- the 'creditTransfer' should then have 4 possible responses if I'm correct: success/duplicate/other fail/no response

Reply:
@jannisvisser Mocks are ready.